src/components/Table/GridTable.tsx

```diff
@@ -294,45 +293,7 @@ export function GridTable<R extends Kinded, S = {}, X extends Only<GridTableXss,
     return rows;
   }, [columns, rows, sorting, sortState]);
 
-  // Calculate the column sizes immediately rather than via the `debounce` method.
-  // We do this for Storybook integrations that may use MockDate. MockDate changes the behavior of `new Date()`, which is used by `useDebounce` and essentially turns off the callback.
-  const calculateImmediately = useRef<boolean>(true);
```

Comment:
I moved all of this code, unchanged, into the `SetupColumnSizes` helper method.

See the comment disclaimer on `SetupColumnSizes` but eslint wants to enforce "hooks can always be called from components", and so we're pretending that `SetupColumnSizes` is a component, vs. one-by-one adding eslint ignore rules to each hook usage in the method.